src/SignalRServiceExtension/Bindings/SignalRAsyncCollector.cs

```diff
@@ -59,13 +63,28 @@ public async Task AddAsync(T item, CancellationToken cancellationToken = default
             else if (convertItem.GetType() == typeof(SignalRGroupAction))
             {
                 SignalRGroupAction groupAction = convertItem as SignalRGroupAction;
-                if (groupAction.Action == GroupAction.Add)
+
+                if (!string.IsNullOrEmpty(groupAction.ConnectionId))
                 {
-                    await client.AddUserToGroup(hubName, groupAction.UserId, groupAction.GroupName).ConfigureAwait(false);
+                    if (groupAction.Action == GroupAction.Add)
+                    {
+                        await client.AddConnectionToGroup(hubName, groupAction.ConnectionId, groupAction.GroupName).ConfigureAwait(false);
+                    }
+                    else
+                    {
+                        await client.RemoveConnectionFromGroup(hubName, groupAction.ConnectionId, groupAction.GroupName).ConfigureAwait(false);
+                    }
                 }
                 else
                 {
-                    await client.RemoveUserFromGroup(hubName, groupAction.UserId, groupAction.GroupName).ConfigureAwait(false);
+                    if (groupAction.Action == GroupAction.Add)
+                    {
+                        await client.AddUserToGroup(hubName, groupAction.UserId, groupAction.GroupName).ConfigureAwait(false);
+                    }
+                    else
+                    {
+                        await client.RemoveUserFromGroup(hubName, groupAction.UserId, groupAction.GroupName).ConfigureAwait(false);
+                    }
                 }
```

Comment:
If both user ID and connection ID are null or empty. The exception will always be `ArgumentException($"{nameof(userId)} cannot be null or empty");` but we don't know which one is actually missing.

Reply:
Discussed with jialin, I want to use a priority order to resolve the parameters. `ConnectionId` -> `UserId` -> `GroupName`. Thus, customer can set both `ConnectionId` and `UserId` and we can update the exception message to something like `connectionId and userId cannot be null or empty at the same time.`